starter_kits/C/src/MyBot.c

```diff
@@ -0,0 +1,324 @@
+// First off, here are some defines and includes. I forget what exactly needs what
+#include <stdio.h>
+#include <stdlib.h>
+
+// We include time.h so we can seed the RNG for default "smart" movement with time()
+#include <time.h>
+
+
+
+// Now for our data structures. These are intentionally left rather basic, so add whatever you want to them
+struct ship{
+	unsigned char x, y;
+	unsigned short halite;
+} ships[4096];
+
+struct player{
+	unsigned char dropoffs[64], dropoff_count, ship_count, ship_yard[2];
+	unsigned short ships[1024];
+	unsigned int halite;
+} players[4];
+
+struct map{
+	unsigned short halite;
+	unsigned short occupied; //(unsigned short)-1 means it's not occupied
+} map[64][64];
+
+
+
+// Now for some global variables to help us keep track of things
+unsigned char dropoffs[256][2], extract_ratio, inspiration_enabled, inspiration_radius, inspiration_ship_count, inspired_extract_ratio, inspired_bonus_multiplier, inspired_move_cost_ratio, mhh, mhw, move_cost_ratio, my_id, map_size[2], number_of_players;
+unsigned short dropoff_cost, max_energy, max_turns, ship_cost, turn_number;
+
+FILE *log_file = NULL;
+
+
+
+// Now let's declare some helper functions
+// You might reasonably want to keep some of these
+
+// Normalize X value (map_width + 1 = 0, 0 - 1 = map_width)
+inline unsigned char norm_x(char x){
+	return (x + map_size[0]) % map_size[0];
+}
+
+// Normalize Y value (map_height + 1 = 0, 0 - 1 = map_height)
+inline unsigned char norm_y(char y){
+	return (y + map_size[1]) % map_size[1];
+}
+
+// Calculate the distance betwen two sets of x and y values
+inline unsigned char calc_dist(unsigned char ax, unsigned char ay, unsigned char bx, unsigned char by){
+	unsigned char dx, dy;
+
+	if(ax > bx)
+		dx = ax - bx;
+	else
+		dx = bx - ax;
+
+	if(ay > by)
+		dy = ay - by;
+	else
+		dy = by - ay;
+
+	return (dx % mhw) + (dy % mhh);
+}
+
+// 0 if not occupied, 1 if occupied
+inline unsigned char tile_occupied(unsigned char x, unsigned char y){
+	return map[x][y].occupied != (unsigned short)-1;
```

Comment:
`(unsigned short)-1` is `65535`... What??

Reply:
Could you add a const called, `UNOCCUPIED` to improve readability.

What IDs can a ship get?

Reply:
The IDs are generated by the game engine. The game engine starts at 0 and adds 1 for every ship generated by all players. Therefore, the max ship ID is theoretically 1999, far below the 65535 unsigned short limit